File: ExamServer/src/main/java/com/nagarro/service/QuizService.java
package com.nagarro.service;

import java.util.List;

import com.nagarro.entity.exam.Category;
import com.nagarro.entity.exam.Quiz;

public interface QuizService {
	Quiz addQuiz(Quiz quiz);

	Quiz updateQuiz(Quiz quiz);

	List<Quiz> getAllQuiz();

	Quiz getQuizById(int id);

	void deleteQuiz(Quiz quiz);

	List<Quiz> getByCategory(Category c);
}

File: Exam-Portal-UI/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { map } from 'rxjs';
import baseUrl from './url';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http: HttpClient) { }

  public getAllQuiz() {
    return this.http.get(`${baseUrl}/quiz/getAll`);
  }

  public addQuiz(quizData: any) {
    return this.http.post(`${baseUrl}/quiz/add`, quizData);
  }

  public deleteQuiz(id: any) {
    return this.http.delete(`${baseUrl}/quiz/delete?v=${id}`);
  }

  public getQuizById(id: any) {
    return this.http.get(`${baseUrl}/quiz/get?v=${id}`);
  }

  public updateQuiz(quizData: any) {
    return this.http.put(`${baseUrl}/quiz/update`, quizData);
  }

  public getQuizByCategory(id: any) {
    return this.http.get(`${baseUrl}/quiz/category?v=${id}`);
  }
}
